### PR TITLE
Add parser support for WITH SUCCESSORS and WITH DESCENDANTS keywords

### PR DESCRIPTION
**Pull request by KOomaXX**
I need to parse ASN1 files that include WITH SUCCESSORS or WITH DESCENDANTS keywords. Currently they are not handled by the parser and compiling those files would crash. 
I've tried to make the minimum modification so that compiling files containing those keyword won't crash.
If you think that there is more to do in order to complete the commit, please explain me what to do. 

Note : Rust code generation tests are failing, but the result of passing the tests ('make test') is the same on my branch and on your master branch so I believe there is no regression.

### CODE REVIEW

File: tests/test_parse.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def test_parse_imports_global_module_reference(self):
                                         'IMPORTS '
                                         'a FROM B '
                                         'c, d FROM E global-module-reference '
-                                        'f, g FROM H {iso(1)}; '
+                                        'f, g FROM H {iso(1)} ;'
```
Comment (eerimoq) [inline tests/test_parse.py:180]: Is this change intended?

Comment (KOomaXX) [reply tests/test_parse.py:180]: Hi,
I don't think so. I guess it's just a typo. I'll correct it.

Comment (KOomaXX) [reply tests/test_parse.py]: Done
